src/NuGet.Clients/PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -511,14 +513,19 @@ private void RefreshAvailableUpdatesCount()
 
             NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
-                var loadContext = new PackageLoadContext(
-                    ActiveSources, Model.IsSolution, Model.Context);
+                var loadContext = new PackageLoadContext(ActiveSources, Model.IsSolution, Model.Context);
                 var packageFeed = await CreatePackageFeedAsync(loadContext, ItemFilter.UpdatesAvailable);
                 var loader = new PackageItemLoader(
                     loadContext, packageFeed, includePrerelease: IncludePrerelease);
 
+                Model.CachedUpdates = new PackageSearchMetadataCache
+                {
+                    Packages = await loader.GetAllPackagesAsync(CancellationToken.None),
+                    IncludePrerelease = IncludePrerelease
+                };
+
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                _topPanel._labelUpgradeAvailable.Count = await loader.GetTotalCountAsync(100, CancellationToken.None);
+                _topPanel._labelUpgradeAvailable.Count = Model.CachedUpdates.Packages.Count;
```

Comment:
this could be null since CachedUpdates is not null checked (in fact you protect for it from being null)